tests/unit/storage/metadata/mongoclient/MongoClientInterface.spec.js

```diff
@@ -846,6 +846,80 @@ describe('MongoClientInterface, tests', () => {
             next => client.deleteBucket(bucketName, logger, err => next(err)),
         ], done);
     });
+
+    describe('MongoClientInterface, putObjectVerCase3 error handling', () => {
+        const bucketName = 'test-bucket-error';
+        let collection;
+
+        beforeEach(done => {
+            createBucket(client, bucketName, true, err => {
+                if (err) {
+                    return done(err);
+                }
+                collection = client.getCollection(bucketName);
+                return done();
+            });
+        });
+
+        afterEach(async () => {
+            await new Promise(resolve =>
+                client.deleteBucket(bucketName, logger, err => {
+                    if (err) {
+                        logger.error('Failed to delete bucket in cleanup', { error: err });
+                    }
+                    resolve();
+                }),
+            );
+        });
```

Comment:
should use callbacks like we do in the beforeEach or make both async ?

Reply:
I kept cb indeed.

Note: I also added a commit for the error handling in mongoclientinterface, it was in the same commit as logging, while it sis a logic change...